proposals/0022-multi-stake.md

```diff
@@ -0,0 +1,181 @@
+---
+simd: '0022'
+title: Multi Stake Account
+authors:
+  - Jon Cinque (Solana Labs)
+category: Standard
+type: Core
+status: Draft
+created: 2023-01-20
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+A new "multi stake" account that allows for multiple movements of stake, useful
+for redelegations, stake pools, and ordinary stakers.
+
+## New Terminology
+
+* "small stake": a stake whose delegation is below the minimum delegation
+lamport amount, currently set to be 1 SOL
+* "multi stake": a stake account with multiple `Delegation` instances
+
+## Motivation
+
+As a minimum stake delegation amount is applied to the network, and potentially
+increased over time, it will become harder for small stakers to participate in
+the network.
+
+Current solutions through stake pools still carry some risk or capital inefficiency
+for the stake pool operator. Either there's a requirement to leave liquid SOL available
+for small stakers to exit, or small stakes cannot enter because there's not enough
+to cover the minimum delegation amount.
+
+Small stakes need to be delegated, while not causing problems in the validator's
+stakes cache.
+
+Side note: this isn't my first proposal for a new stake account type. Here's an earlier
+attempt meant for redelegation: https://github.com/solana-labs/solana/pull/24762
+
+## Alternatives Considered
+
+The easiest solution is to change the validator's stakes cache, since that only
+impacts the specific implementation in the Solana Labs validator.
+
+In the perfect situation, small stakes are properly delegated, but don't receive
+rewards. Meaning:
+
+* they don't take up space in the stakes cache
+* they take the correct amount of epoch boundaries to activate / deactivate
+* they are included in a validator's voting power
+* they can be slashed
+
+There are a few ways of solving this at the stakes cache level, and they all have
+serious issues.
+
+* Include small stakes in the cache, but don't pay out rewards
+
+This is the simplest solution, which essentially changes "minimum delegation amount"
+to mean "minimum reward earning amount". So you can delegate stake accounts with
+less than the minimum delegation, but they won't earn rewards.
+
+Unfortunately, small stakes can still bloat the stakes cache from a memory level.
+
+* Don't include small stakes in the cache, but track them in transactions
+
+In this solution, on every transaction, if the runtime sees a small stake, it
+includes its delegation amount in the validator's voting power, but doesn't
+add its pubkey and `Delegation` to the cache.
+
+The runtime tracks the pre and post state of every account, and for a successful
+transaction, subtracts the old small stake delegation amount, and adds the new small
+stake delegation amount.
+
+During rewards payout, small stakes are completely omitted, since they are not
+present in the cache.
+
+During a slashing event, it's still possible to query it using `get_program_accounts`.
```

Comment:
Very minor nit: I don't think it's helpful to mention slashing here at all. There's a lot of competing and conflicting thoughts about exactly what slashing will/might/wont be and who is at risk. Introducing it here just produces unhelpful comments that are relevant to this targeted proposal (this comment included!). All stake accounts are still on chain (obviously) and some future big bad slasher could certainly 🔪  'em

Reply:
Yeah good point, I'll just remove the line. Especially considering it's for a weak alternative.